contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -428,6 +428,15 @@ It should be included as a top level element in status, similar to
 Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```

Comment:
do we need the listType / listMapKey comments in built-in types as well, or is the patchStrategy/patchMergeKey inherited by SSA?

Reply:
"If listType is missing, the API server interprets a patchMergeStrategy=merge marker as a listType=map and the corresponding patchMergeKey marker as a listMapKey." ([xref](https://kubernetes.io/docs/reference/using-api/server-side-apply/))

That said, it looks like Conditions for built-in types are often defined as:

```
         // +listType=map                                                                                                           
         // +listMapKey=type
```

I can update this doc for that as well.

Reply:
yeah, if we do that, that works for built-in and kubebuilder types, so we can drop the kubebuilder language

Reply:
@apelisse is there any reason to recommend `+patchStrategy` and `+patchMergeKey` on new built-in types? I see a mix of "only patchStrategy/patchMergeKey", "both patchStrategy/patchMergeKey and SSA listType/listMapKey" and "only  SSA listType/listMapKey" in the built-in types.

Reply:
simplified to just use SSA in the built-in example form

Reply:
Well… don't we want CSA to still work?

Reply:
Found another patch strategy ref in the doc. I've updated that too.

Reply:
> Well… don't we want CSA to still work?

I'm not sure.  We have a couple cases like: https://github.com/kubernetes/kubernetes/blob/f8b5f1a77bbab217159b1d024000770f8c95ed97/staging/src/k8s.io/api/flowcontrol/v1beta2/types.go#L360

Reply:
Should we audit all of this?   I'm seeing a wide range of annotation patterns. There are some surprising things in here like [JobCondition](https://github.com/kubernetes/kubernetes/blob/f8b5f1a77bbab217159b1d024000770f8c95ed97/staging/src/k8s.io/api/batch/v1/types.go#L421-L425) where SSA uses atomic.

Reply:
yeah, a sweep is probably good